run.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# This script is used to run the main script in a virtual environment.
+export PATH="/usr/local/bin:$PATH" # Add /usr/local/bin to the PATH variable
+
+set_venv() {
+    if [ -d "venv" ]; then # Check if the virtual environment exists
+        source venv/bin/activate # Activate the virtual environment
+    else
+        echo "venv is not found. Creating venv..."
+        create_venv
+    fi
+}
+
+create_venv() {
+    python3 -m venv venv && # Create a virtual environment
+    source venv/bin/activate # Activate the virtual environment
+}
+
+{ # try
+    set_venv # Set the virtual environment
+} || { # catch
+    echo "An error occurred while creating the virtual environment."
+    echo "venv is not found. Installing venv..."
+    sudo apt-get update
```

Comment:
When I run something I don’t expect that system packages will be installed. Also what about other distributions which are not using apt for package management?